DDSim/DDSim/DD4hepSimulation.py

```diff
@@ -636,6 +654,52 @@ def __addParametersToRunHeader( self ):
 
     return runHeader
 
+
+  def __addParametersToEvent(self, parameters):
+    """ extract event parameters from the command line, to write in the header
+    of each event.
+    
+    :param parameters: list of command line parameters to parse
+    """
+    print parameters
+    stringParameters, intParameters, floatParameters, allParameters = {}, {}, {}, []
+    for p in parameters:
+      parameterAndValue = p.split( "=", 1 )
+      if len(parameterAndValue) is not 2:
+        self._errorMessages.append("ERROR: Couldn't decode event parameter '%s'" %(p))
+        raise SyntaxError("ERROR: Couldn't decode event parameter '%s'" %(p))
```

Comment:
This whole function comes after the errorMessages were used up, so it is not necessary to store things in the error messages. Or add another check for error messages after this function was called.
So two possibilities:
1) remove the appending to errorMessages and remove the `continue` (because the code after `raise` is never executed)
2) remove the `raise` and add a check for error messages after the function was called

Same goes for the other places where errorMessages/exceptions are used.